module/mempool/stdmap/incorporated_result_seals.go

```diff
@@ -46,15 +129,66 @@ func (ir *IncorporatedResultSeals) ByID(id flow.Identifier) (*flow.IncorporatedR
 
 // Rem removes an IncorporatedResultSeal from the mempool
 func (ir *IncorporatedResultSeals) Rem(id flow.Identifier) bool {
-	return ir.Backend.Rem(id)
+	removed := false
+	err := ir.Backend.Run(func(entities map[flow.Identifier]flow.Entity) error {
+		var entity flow.Entity
+		entity, removed = ir.Backdata.Rem(id)
+		if !removed {
+			return nil
+		}
+		seal := entity.(*flow.IncorporatedResultSeal)
+		ir.removeFromIndex(id, seal.Header.Height)
+		return nil
+	})
+	if err != nil {
+		panic(err)
+	}
+	return removed
 }
 
-// Clear removes all entities from the pool.
 func (ir *IncorporatedResultSeals) Clear() {
-	ir.Backend.Clear()
+	err := ir.Backend.Run(func(_ map[flow.Identifier]flow.Entity) error {
+		ir.Backdata.Clear()
+		ir.byHeight = make(map[uint64]sealSet)
+		return nil
+	})
+	if err != nil {
+		panic(err)
+	}
 }
 
 // RegisterEjectionCallbacks adds the provided OnEjection callbacks
 func (ir *IncorporatedResultSeals) RegisterEjectionCallbacks(callbacks ...mempool.OnEjection) {
 	ir.Backend.RegisterEjectionCallbacks(callbacks...)
 }
+
+// PruneUpToHeight remove all seals for blocks whose height is strictly
+// smaller that height. Note: seals for blocks at height are retained.
+func (ir *IncorporatedResultSeals) PruneUpToHeight(height uint64) error {
+	return ir.Backend.Run(func(entities map[flow.Identifier]flow.Entity) error {
+		if height < ir.lowestHeight {
+			return fmt.Errorf("new pruning height %v cannot be smaller than previous pruned height:%v", height, ir.lowestHeight)
+		}
+
+		if len(entities) == 0 {
+			ir.lowestHeight = height
+			return nil
+		}
```

Comment:
Just checking this assumption is OK to embed in this layer: once we prune up to a height, we can never insert seals for lower heights

Reply:
Yes, this is intended. You are right, we should be documenting it 🙏